python_ta/checkers/constant_test_checker.py

```diff
@@ -0,0 +1,55 @@
+"""
+checker for using constant value in a conditional statement
+"""
+import astroid
+
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+
+class UsingConstantTestChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+
+    name = 'using-constants-test'
+
+    msgs = {'W7777': ('using a constant value in a conditional statement'
+                      , 'using-constants-test'
+                      , 'Conditional statements should depend on a variable not a constant value.'
+                        'This is usually not what the user intended to do'),}
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    def _check_all_constants(self, node):
+        """
+        Precondition: node is a condition in an if statement
+        Returns true if all values in the BinOp tree are constants or if all values in the UnaryOp tree are constants
```

Comment:
This should be more concise (no need to talk about different types of notes). Something like "all values in this node are constants"